easybuild/easyblocks/o/openblas.py

```diff
@@ -39,6 +70,38 @@ def configure_step(self):
             # There doesn't seem to be a POWER9 option yet, but POWER8 should work.
             print_warning("OpenBLAS 0.3.5 and lower have known issues on POWER systems")
             default_opts[TARGET] = 'POWER8'
+        elif compiler_optarch:
+            compiler_family = self.toolchain.comp_family()
+            self.log.info("EasyBuild full optarch requested for %s: %s" % (compiler_family, compiler_optarch))
+            optarch_as_target = self._parse_optarch(compiler_optarch)
+            mapped_target = None
+
+            if optarch_as_target:
+                # Note that _parse_optarch returns lowercased results, so GENERIC has become 'generic'.
+                if optarch_as_target == 'generic':
+                    self._set_dynamic_architecture(default_opts)
+                    mapped_target = 'generic'
+                else:
+                    self.log.info("EasyBuild march: %s" % optarch_as_target)
+                    openblas_targets = self._get_openblas_targets(self.cfg['targetfile'])
+                    mapped_target = self._get_mapped_target(optarch_as_target, openblas_targets)
+            else:
+                self.log.info("Optarch specified for %s, but no march detected", compiler_family)
+
+            if mapped_target is None:
+                print_warning("optarch for %s given as '%s'\n"
+                              "EasyBuild was unable to map this to an equivalent OpenBLAS target!\n"
+                              "OpenBLAS will be built to optimize its kernel at runtime!\n"
+                              % (self.toolchain.comp_family(), compiler_optarch))
+                self.log.warning("Unable to map %s to an OpenBLAS target, falling back to runtime optimization."
+                                 % compiler_optarch)
+                self._set_dynamic_architecture(default_opts)
```

Comment:
I'm very unsure about this. If I build OpenBLAS currently, it seems to autodetect the current arch and optimizes for it. It seems to even overwrite our optarch settings etc which is fine when building for native. Additionally e.g. for POWER and OpenBLAS until 0.3.7 (IIRC) DYNAMIC_ARCH is broken to the point where even the simple tests fail (so in this case the EC wouldn't build)

Reply:
We should certainly disable DYNAMIC_ARCH for Power. But the thing is, if you ask for `optarch=haswell` and OpenBLAS optimises for native, and you're on a skylake build host, your compute nodes will cease to function. For me (and I presume others), `optarch` is not something that may be ignored.

My strong preference is that if an EasyBuild user sets `optarch=haswell` EasyBuild offers the guarantee that the code produced will indeed run on haswell.

Reply:
Alright sure. I just wanted to highlight that the default (native) should still work as before (hence my restriction to "when building for native")